src/agent.spec.ts

```diff
@@ -0,0 +1,236 @@
+    mockProvider.addCallTo(MOCK_DATA2.poolAddress, 1, Ipool, "token1", { inputs: [], outputs: [MOCK_DATA2.token1] });
+    mockProvider.addCallTo(MOCK_DATA2.poolAddress, 1, Ipool, "fee", {
+      inputs: [],
+      outputs: [BigNumber.from(MOCK_DATA2.fee)],
+    });
+
+    const txEvent = new TestTransactionEvent()
+      .setFrom(MOCK_DATA2.from)
+      .setTo(MOCK_DATA2.to)
+      .setBlock(1)
+      .addEventLog(UNISWAP_V3_POOL_ADDR[0], MOCK_DATA2.poolAddress, [
+        MOCK_DATA2.sender,
+        MOCK_DATA2.recipient,
+        MOCK_DATA2.amount0,
+        MOCK_DATA2.amount1,
+        MOCK_DATA2.sqrtPriceX96,
+        MOCK_DATA2.liquidity,
+        MOCK_DATA2.tick,
+      ]);
+    mockProvider.setLatestBlock(1);
+    const findings = await handleTransaction(txEvent);
+
+    expect(findings.length).toEqual(0);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns multiple findings from different pools of uniswap", async () => {
+    mockProvider.addCallTo(MOCK_DATA.poolAddress, 0, Ipool, "token0", { inputs: [], outputs: [MOCK_DATA.token0] });
+    mockProvider.addCallTo(MOCK_DATA.poolAddress, 0, Ipool, "token1", { inputs: [], outputs: [MOCK_DATA.token1] });
+    mockProvider.addCallTo(MOCK_DATA.poolAddress, 0, Ipool, "fee", {
+      inputs: [],
+      outputs: [BigNumber.from(MOCK_DATA.fee)],
+    });
+    mockProvider.addCallTo(MOCK_DATA1.poolAddress, 0, Ipool, "token0", { inputs: [], outputs: [MOCK_DATA1.token0] });
+    mockProvider.addCallTo(MOCK_DATA1.poolAddress, 0, Ipool, "token1", { inputs: [], outputs: [MOCK_DATA1.token1] });
+    mockProvider.addCallTo(MOCK_DATA1.poolAddress, 0, Ipool, "fee", {
+      inputs: [],
+      outputs: [BigNumber.from(MOCK_DATA1.fee)],
+    });
+
+    const txEvent = new TestTransactionEvent()
+      .setBlock(0)
+      .addEventLog(UNISWAP_V3_POOL_ADDR[0], MOCK_DATA.poolAddress, [
+        MOCK_DATA.sender,
+        MOCK_DATA.recipient,
+        MOCK_DATA.amount0,
+        MOCK_DATA.amount1,
+        MOCK_DATA.sqrtPriceX96,
+        MOCK_DATA.liquidity,
+        MOCK_DATA.tick,
+      ])
+      .addEventLog(UNISWAP_V3_POOL_ADDR[0], MOCK_DATA1.poolAddress, [
+        MOCK_DATA1.sender,
+        MOCK_DATA1.recipient,
+        MOCK_DATA1.amount0,
+        MOCK_DATA1.amount1,
+        MOCK_DATA1.sqrtPriceX96,
+        MOCK_DATA1.liquidity,
+        MOCK_DATA1.tick,
+      ]);
+
+    mockProvider.setLatestBlock(0);
+
+    const MOCK_FINDING1 = {
+      name: "Uniswap V3 Swap Event",
+      description: `swap event detected in uniswap v3`,
+      alertId: "UNISWAP-V3-SWAP-EVENT",
+      protocol: "Uniswap",
+      severity: FindingSeverity.Info,
+      type: FindingType.Info,
+      metadata: {
+        poolAddress: MOCK_DATA.poolAddress,
+        sender: MOCK_DATA.sender,
+        recipient: MOCK_DATA.recipient,
+        amount0: MOCK_DATA.amount0.toString(),
+        amount1: MOCK_DATA.amount1.toString(),
+      },
+      addresses: [],
+      labels: [],
+    };
+
+    const MOCK_FINDING2 = {
```

Comment:
these findings in tests repeat themselves so you could get rid of them and instead create a function like `createFinding` and pass mock data into it.